libusb/hid.c

```diff
@@ -1582,8 +1666,31 @@ int HID_API_EXPORT hid_send_feature_report(hid_device *dev, const unsigned char
 		(unsigned char *)data, length,
 		1000/*timeout millis*/);
 
-	if (res < 0)
+	if (res < 0) {
+		wchar_t *custom_string = NULL;
+
+		switch (res) {
+		case LIBUSB_ERROR_TIMEOUT:
+			custom_string = L"Transfer timed out";
+			break;
+		case LIBUSB_ERROR_PIPE:
+			custom_string = L"Control request not supported by device";
+			break;
+		case LIBUSB_ERROR_NO_DEVICE:
+			custom_string = L"Device has disconnected";
+			break;
+		case LIBUSB_ERROR_BUSY:
+			custom_string = L"Called from event handling context";
+			break;
+		case LIBUSB_ERROR_INVALID_PARAM:
+			custom_string = L"Transfer size larger than supported";
```

Comment:
that's not nesessary true

Reply:
It is what the libusb documentation says though.

```
libusb_control_transfer()

Returns... LIBUSB_ERROR_INVALID_PARAM if the transfer size is larger than the operating system and/or hardware can support (see Transfer length limitations)
```

If it's not true, then the libusb documentation should be updated.